PyPDF2/_reader.py

```diff
@@ -1966,7 +1971,7 @@ def xfa(self) -> Optional[Dict[str, Any]]:
 
 class PdfFileReader(PdfReader):  # pragma: no cover
     def __init__(self, *args: Any, **kwargs: Any) -> None:
-        deprecate_with_replacement("PdfFileReader", "PdfReader")
+        deprecation_with_replacement("PdfFileReader", "PdfReader", "3.0.0")
```

Comment:
As opposed to having this continue to work just fine, but having a message say that "This was removed in 3.0.0", maybe better to just have this hard error, forcing the user to change? Otherwise, feels weird to have a deprecation message saying some functionality was removed in `3.0.0` and wasn't, and would read to me as a bug of some sort.

Reply:
The `deprecation_with_replacement` raises an exception instead of using `warnings.warn`. So it will stop working.

The point that I want to address by leaving it here is that (1) many people don't have warnings enabled and (2) a lot of deprecated code still gets written to new tutorials / answers on StackExchange. I see PdfFileParse all the time.

Reply:
I had two ideas in mind when I didn't remove the function body:

1. The diff is a little bit cleaner. It is still very messy, though.
2. If anybody really needs it, they could overwrite the deprecation_X functions and the code would still work
3. I'm looking forward to a big commit where we remove hundreds of lines :smile: 

Those arguments are pretty weak, though.

> I do think having a second release of more strongly removing them while still having it be callable makes sense given the very long tail of Internet tutorials, and also using the opportunity of pypdf to remove it since usage of the new package should be a good separation signal from outdated guides.

I'm not sure if I can follow. What do you think how we should proceed (with this PR and the versions of PyPDF2/pypdf)?